velox/functions/sparksql/tests/RegexFunctionsTest.cpp

```diff
@@ -41,6 +43,223 @@ class RegexFunctionsTest : public test::SparkFunctionBaseTest {
     return evaluateOnce<std::string>(
         fmt::format("regexp_extract(c0, '{}')", pattern), str);
   }
+
+  void testRegexReplaceSimpleOnce(
+      const std::vector<std::optional<std::string>>& input,
+      std::string pattern,
+      std::string replace,
+      std::string output,
+      std::optional<int> position =
+          std::nullopt /*adding the optional position parameter*/) {
+    auto valueAt = [&input](vector_size_t row) {
+      return input[row] ? StringView(*input[row]) : StringView();
+    };
+    auto nullAt = [&input](vector_size_t row) {
+      return !input[row].has_value();
+    };
+    auto result = [&] {
+      auto inputString =
+          makeFlatVector<StringView>(input.size(), valueAt, nullAt);
+      auto rowVector = makeRowVector({inputString});
+
+      if (!position) {
+        return evaluateOnce<std::string>(
+            fmt::format("regex_replace(c0, '{}', '{}')", pattern, replace),
+            rowVector);
+      } else {
+        return evaluateOnce<std::string>(
+            fmt::format(
+                "regex_replace(c0, '{}', '{}', {})",
+                pattern,
+                replace,
+                *position),
+            rowVector);
+      }
+    }();
+
+    EXPECT_EQ(result, output);
```

Comment:
It would be better to move this check to the caller. Have this function return result and have the caller call EXPECT_EQ. This way the code will be more readable and the failure will point to the line in the test that failed.

Reply:
Fixed with new method of testing in latest commit.